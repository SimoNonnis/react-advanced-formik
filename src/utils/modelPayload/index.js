import forIn from 'lodash/forIn';
import trim from 'lodash/trim';

function modelPayload(obj) {
  const newObj = {};

  forIn(obj, (value, key) => {
    switch (value) {
      case '':
      case null:
        newObj[key] = null;
        break;
      case true:
        newObj[key] = true;
        break;
      case false:
        newObj[key] = false;
        break;
      case 'true':
        newObj[key] = true;
        break;
      case 'false':
        newObj[key] = false;
        break;
      default:
        newObj[key] = trim(value);
    }
  });

  delete newObj.primary;
  delete newObj.living_at_the_property_you_are_selling_for_at_least_twelve_months;

  return newObj;
}

export default modelPayload;
