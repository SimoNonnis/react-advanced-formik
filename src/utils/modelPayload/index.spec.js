/* eslint-env jest */
import modelPayload from './';

describe('Test modelPayload()', () => {
  it('should transform string value "true" to boolean true', () => {
    expect(modelPayload({ test: 'true' })).toMatchSnapshot();
  });
  it('should transform string value "false" to boolean false', () => {
    expect(modelPayload({ test: 'false' })).toMatchSnapshot();
  });
  it('should transform empty string value to null', () => {
    expect(modelPayload({ test: '' })).toMatchSnapshot();
  });
  it('should leave the value unchanged', () => {
    expect(modelPayload({ test: 'test string' })).toMatchSnapshot();
  });
  it('should leave true unchanged', () => {
    expect(modelPayload({ test: true })).toMatchSnapshot();
  });
  it('should leave false unchanged', () => {
    expect(modelPayload({ test: false })).toMatchSnapshot();
  });
  it('should leave null unchanged', () => {
    expect(modelPayload({ test: null })).toMatchSnapshot();
  });
  it('should leave null unchanged', () => {
    expect(modelPayload({ test: null })).toMatchSnapshot();
  });
  it('should delete primary key', () => {
    expect(modelPayload({ primary: true })).toMatchSnapshot();
  });
  it('should delete living_at_the_property_you_are_selling_for_at_least_twelve_months key', () => {
    expect(
      modelPayload({ living_at_the_property_you_are_selling_for_at_least_twelve_months: true })
    ).toMatchSnapshot();
  });
});
