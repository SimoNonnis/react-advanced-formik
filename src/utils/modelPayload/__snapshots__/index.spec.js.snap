// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test modelPayload() should delete living_at_the_property_you_are_selling_for_at_least_twelve_months key 1`] = `Object {}`;

exports[`Test modelPayload() should delete primary key 1`] = `Object {}`;

exports[`Test modelPayload() should leave false unchanged 1`] = `
Object {
  "test": false,
}
`;

exports[`Test modelPayload() should leave null unchanged 1`] = `
Object {
  "test": null,
}
`;

exports[`Test modelPayload() should leave null unchanged 2`] = `
Object {
  "test": null,
}
`;

exports[`Test modelPayload() should leave the value unchanged 1`] = `
Object {
  "test": "test string",
}
`;

exports[`Test modelPayload() should leave true unchanged 1`] = `
Object {
  "test": true,
}
`;

exports[`Test modelPayload() should transform empty string value to null 1`] = `
Object {
  "test": null,
}
`;

exports[`Test modelPayload() should transform string value "false" to boolean false 1`] = `
Object {
  "test": false,
}
`;

exports[`Test modelPayload() should transform string value "true" to boolean true 1`] = `
Object {
  "test": true,
}
`;
